cmd/sandbox-ce/cmd/config.go

```diff
@@ -0,0 +1,216 @@
+
+type configCmd struct {
+	cli.PlainCommand `name:"config" short-description:"Config"`
+}
+
+type configDownloadCmd struct {
+	Command `name:"download" short-description:"Download"`
+
+	Revision string `short:"r" long:"revision" description:"revision to download" default:"master"`
+	URL      string `short:"u" long:"url" description:"url of the config to download"`
+}
+
+func (c *configDownloadCmd) Execute(args []string) error {
+	var err error
+	var confURL, outPath string
+
+	if c.URL != "" {
+		confURL = c.URL
+		outPath, err = getConfigPath(strconv.Itoa(int(time.Now().Unix())))
+	} else {
+		confURL = getComposerConfFileURL(c.Revision)
+		outPath, err = getConfigPath(c.Revision)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	err = c.downloadConfig(confURL, outPath)
+	if err != nil {
+		return err
+	}
+
+	return setDefaultConfig(outPath)
+}
+
+func (c *configDownloadCmd) downloadConfig(url, path string) error {
+	out, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	return err
+}
+
+type configListCmd struct {
+	Command `name:"list" short-description:"List"`
+}
+
+func (c *configListCmd) Execute(args []string) error {
+	confDirPath, err := getConfigDirPath()
+	if err != nil {
+		return err
+	}
+
+	if _, err := os.Stat(confDirPath); err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	files, err := ioutil.ReadDir(confDirPath)
+	if err != nil {
+		return err
+	}
+
+	defaultRevision, err := getDefaultConfigRevision()
+	if err != nil {
+		return err
+	}
+
+	for _, f := range files {
+		entry := f.Name()
+		if entry == defaultConfigName {
+			continue
+		}
+
+		if entry == defaultRevision {
+			entry = fmt.Sprintf("%s (default)", entry)
+		}
+
+		fmt.Println(entry)
+	}
+
+	return nil
+}
+
+type configSetDefaultCmd struct {
+	Command `name:"set-default" short-description:"Set default config"`
+
+	Args struct {
+		Revision string `positional-arg-name:"revision"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+func (c *configSetDefaultCmd) Execute(args []string) error {
+	revisionConfPath, err := getConfigPath(c.Args.Revision)
+	if err != nil {
+		return err
+	}
+
+	if _, err := os.Stat(revisionConfPath); err != nil {
+		if os.IsNotExist(err) {
+			return errors.Wrapf(err, "no configuration found for revision `%s`",
+				c.Args.Revision)
+		}
+
+		return err
+	}
+
+	return setDefaultConfig(revisionConfPath)
+}
+
+func setDefaultConfig(confPath string) error {
+	defaultConfPath, err := getConfigPath(defaultConfigName)
+	if err != nil {
+		return err
+	}
+
+	if _, err := os.Lstat(defaultConfPath); err == nil {
+		if err := os.Remove(defaultConfPath); err != nil {
+			return fmt.Errorf("failed to unlink: %+v", err)
+		}
+	}
+
+	return os.Symlink(confPath, defaultConfPath)
+}
+
+func getConfigDirPath() (string, error) {
```

Comment:
similar here. Let's abstract this code to reuse in both commands. And in new "update" command also.